compress_string.cpp

```diff
@@ -1,41 +1,34 @@
 #include <iostream>
-#include <vector>
-#include <string>
 #include <sstream>
+#include <string>
+#include <vector>
 
 /*
   Given a pointer to a string to compress
   return a string in the form:
   <char><char_count> of all the characters in the to_compress string
   eg. aaabbcccdff -> a3b2c3d1f2
 
+
   clang++-3.7 -std=c++11 compress_string.cpp -o compress_string
   ./compress_string <string_to_compress>
 */
 
-using namespace std;
-
-string compress(const string &to_compress)
+std::string compress(const std::string &to_compress)
 {
-	/*
-	  Takes a reference (not pointer) to a const string, which makes the
-	  string immutable. Creates a counter (init at 0) and a stringstream,
-	  which is initialised with the first character of the string. After
-	  that, iterate over the given string compare it to the last char in the
-	  stringstream. If the char is different, append the counter of the
-	  previous char, reset counter to 1 and append the new char.
-
-	 */
-	stringstream ss;
-	int counter = 0;
-	ss << to_compress.at(0);
+	int count = 0;
+	char letter = 0; // not a letter
+
+	std::stringstream ss;
 	for (char c : to_compress) {
-		if (ss.str().back() == c) {
-			++counter;
+		if (c == letter) {
+			++count;
 		} else {
-			// new element - add to letters and set its count to 1
-			ss << counter;
-			counter = 1;
+			if (count > 1) {
```

Comment:
In fact, I don't see much use in this check. We append the `count` to the stringstream whatever it is. The minimum possible value is 1, so we should check `count > 0`, but that seems unnecessary to me